Sources/ForageSDK/Foundation/Network/LiveForageService.swift

```diff
@@ -52,76 +52,90 @@ internal class LiveForageService: ForageService {
-                            sessionToken: request.authorization,
-                            merchantID: request.merchantID,
-                            paymentMethodRef: request.paymentMethodReference,
-                            completion: { paymentMethodResult in
-                                switch paymentMethodResult {
-                                case .success(let paymentMethod):
-                                    if (paymentMethod.balance == nil) {
-                                        let forageError = ForageError(errors:[ForageErrorObj(httpStatusCode: 500, code:"invalid_data", message:"Invalid Data")])
-                                        self?.logger?.error(
-                                            "Balance check failed for Payment Method \(paymentMethodRef). Balance not attached",
-                                            error: forageError,
-                                            attributes: nil
-                                        )
-                                        
-                                        completion(.failure(forageError))
-                                        return
-                                    }
-                                    self?.logger?.notice(
-                                        "Balance check succeeded for Payment Method \(paymentMethodRef)",
-                                        attributes: nil
-                                    )
-                                    completion(.success(paymentMethod.balance!))
-                                case .failure(let error):
-                                    completion(.failure(error))
-                                }
-                                
-                            }
-                        )
-                    case .failure(let error):
-                        self?.logger?.error(
-                            "Balance check failed for Payment Method \(paymentMethodRef)",
-                            error: error,
-                            attributes: nil
-                        )
-                        completion(.failure(error))
-                    }
-                })
+        do {
+            // TODO: parallelize the first 2 requests!
+            
+            // 1. Prologue: checkBalance - get the Encryption Key and the Payment Method
+            let xKeyModel = try await awaitResult { completion in
+                self.getXKey(sessionToken: sessionToken, merchantID: merchantID, completion: completion)
             }
+            let paymentMethod = try await awaitResult { completion in
+                self.getPaymentMethod(
+                    sessionToken: sessionToken,
+                    merchantID: merchantID,
+                    paymentMethodRef: paymentMethodReference,
+                    completion: completion
+                )
+            }
+            
+            let balanceRequest = ForageRequestModel(
```

Comment:
Just calling out that I think the `ForageRequestModel` is silly. We should have specific request models for specific requests rather than defaulting things to the empty string. This decision was made Symphony a long long time ago.

Reply:
agreed, will refactor!